Tests/Unit/Content/Application/ContentMerger/ContentMergerTest.php

```diff
@@ -45,40 +45,45 @@ public function testMerge(): void
         ]);
 
         $mergedDimensionContent = $this->prophesize(DimensionContentInterface::class);
-        $dimensionContent1 = $this->prophesize(DimensionContentInterface::class);
-        $dimensionContent2 = $this->prophesize(DimensionContentInterface::class);
-        $dimensionContent3 = $this->prophesize(DimensionContentInterface::class);
 
-        $mostSpecificDimension = $this->prophesize(DimensionInterface::class);
-
-        $resource = $this->prophesize(ContentRichEntityInterface::class);
-        $resource->createDimensionContent($mostSpecificDimension->reveal())
+        $resource = $this->prophesize(Example::class);
+        $resource->createDimensionContent()
             ->willReturn($mergedDimensionContent->reveal());
 
-        $dimensionContent3->getDimension()->willReturn($mostSpecificDimension->reveal());
-        $dimensionContent3->getResource()->willReturn($resource->reveal());
-
-        $merger1->merge($mergedDimensionContent->reveal(), $dimensionContent1->reveal())->shouldBeCalled();
-        $merger2->merge($mergedDimensionContent->reveal(), $dimensionContent1->reveal())->shouldBeCalled();
+        $dimensionContent1 = new ExampleDimensionContent($resource->reveal());
+        $dimensionContent1->setLocale(null);
+        $dimensionContent1->setStage('draft');
+        $dimensionContent2 = new ExampleDimensionContent($resource->reveal());
+        $dimensionContent1->setLocale('en');
+        $dimensionContent1->setStage('draft');
 
-        $merger1->merge($mergedDimensionContent->reveal(), $dimensionContent2->reveal())->shouldBeCalled();
-        $merger2->merge($mergedDimensionContent->reveal(), $dimensionContent2->reveal())->shouldBeCalled();
+        $merger1->merge($mergedDimensionContent->reveal(), $dimensionContent1)->shouldBeCalled();
+        $merger2->merge($mergedDimensionContent->reveal(), $dimensionContent1)->shouldBeCalled();
 
-        $merger1->merge($mergedDimensionContent->reveal(), $dimensionContent3->reveal())->shouldBeCalled();
-        $merger2->merge($mergedDimensionContent->reveal(), $dimensionContent3->reveal())->shouldBeCalled();
+        $merger1->merge($mergedDimensionContent->reveal(), $dimensionContent2)->shouldBeCalled();
+        $merger2->merge($mergedDimensionContent->reveal(), $dimensionContent2)->shouldBeCalled();
 
-        $mergedDimensionContent->markAsMerged()->shouldBeCalled();
+        $mergedDimensionContent->setLocale(null) // TODO FIXME find a way to avoid this call (ExampleControllerTest::testPostTriggerUnpublish) currently files without this
```

Comment:
```suggestion
        $mergedDimensionContent->setLocale(null) // TODO FIXME find a way to avoid this call (ExampleControllerTest::testPostTriggerUnpublish) currently fails without this
```

think the `ExampleControllerTest::testPostTriggerUnpublish` fails because of https://github.com/sulu/SuluContentBundle/issues/145